### ISSUE TITLE
Suggest `-z`/`-n` for comparison against empty string (`= ''`/`!= ''`)

### ISSUE THREAD

**Issue by e-kwsm** [koalaman/shellcheck#3187]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2244?
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related #1664?

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
# shellcheck enable=avoid-nullary-conditions
[ "$1" = '' ]
[ "$1" != '' ]
```

#### Here's what shellcheck currently says:

Nothing.

#### Here's what I wanted or expected to see:

```

In - line 3:
[ "$1" = '' ]
  ^-------^ SCXXXX (style): Prefer -z to check empty string

Did you mean:
[ -z "$1" ]


In - line 4:
[ "$1" != '' ]
  ^--------^ SCXXXX (style): Prefer -n to check non-empty string

Did you mean:
[ -n "$1" ]

```
